### PR TITLE
Dark Mode - NxFilterInput clear button hover - RSC-1563

### PR DESCRIPTION
**Pull request by avnvbhatta**
https://issues.sonatype.org/browse/RSC-1563

### CODE REVIEW

File: lib/src/components/NxFilterInput/NxFilterInput.scss [hunk 1]
```diff
@@ -38,7 +38,7 @@
     @include nx-button-helpers.small-icon-btn;
 
     margin: 0;
-    padding: calc(var(--nx-spacing-base) - 2px) 0;
+    padding: calc(var(--nx-spacing-base) - var(--nx-padding-offset-filter-clear-button)) 0;
```
Comment (achen5) [inline lib/src/components/NxFilterInput/NxFilterInput.scss:41]: This works, but do you mind explaining the purpose of this calculation? Perhaps it's simpler to do the following?
```
--nx-padding-filter-clear-button: 2px;

@include nx-dark-mode-helpers.dark-mode {
    --nx-padding-filter-clear-button: 1px;
}
```
```
padding: var(--nx-padding-filter-clear-button) 0;
```

File: gallery/src/styles/CssVariables/CssVariablesPage.tsx [hunk 2]
```diff
@@ -236,6 +236,10 @@ const CssVariablesPage = () => {
           <PropertyDocItem propertyVar="--nx-border-grid">
             The standard border style for <NxCode>nx-grid</NxCode> vertical borders.
           </PropertyDocItem>
+          <PropertyDocItem propertyVar="--nx-padding-offset-filter-clear-button">
+            The padding offset used to calculate the correct height of the clear button
+            in <NxCode>NxFilterInput</NxCode>.
+          </PropertyDocItem>
```
Comment (achen5) [inline gallery/src/styles/CssVariables/CssVariablesPage.tsx:242]: image
Looks like the gallery is having issues rendering this row. You may need to widen the first column.

File: lib/src/components/NxFilterInput/NxFilterInput.scss (hunk 1 above)
Comment (rpokorny) [inline lib/src/components/NxFilterInput/NxFilterInput.scss:41]: I don't think we need a new variable here.  We just need to take one additional pixel off of the top and bottom paddings produced by `nx-button-helpers.small-icon-btn`.  How about giving that mixin a parameter called `extra-vertical-padding` or something, and then passing `-1px` for it?

Comment (avnvbhatta) [reply lib/src/components/NxFilterInput/NxFilterInput.scss]: > I don't think we need a new variable here. We just need to take one additional pixel off of the top and bottom paddings produced by `nx-button-helpers.small-icon-btn`. How about giving that mixin a parameter called `extra-vertical-padding` or something, and then passing `-1px` for it?

Thank you, that's a great idea. Changed here: ab33a4d82a32620ec803fe44ebbea22d815d66ca and e7eeb6339c172b9ffa6ae94b9fb9cc35f2058bcb

File: gallery/src/styles/CssVariables/CssVariablesPage.tsx (hunk 2 above)
Comment (avnvbhatta) [reply gallery/src/styles/CssVariables/CssVariablesPage.tsx]: > image Looks like the gallery did not render this row correctly. You may need to widen the first column.

Removed: e7eeb6339c172b9ffa6ae94b9fb9cc35f2058bcb

**Pull request merged**